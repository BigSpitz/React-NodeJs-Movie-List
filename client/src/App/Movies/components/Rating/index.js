import React from "react";
import { CircularProgress, Typography, Box } from "@material-ui/core";
import styled from "styled-components";

const StyledRatingContainer = styled.div`
  width: 50px;
  height: 50px;
  background-color: black;
  border-radius: 50%;
  color: white;
`;

const Rating = ({ rating }) => {
  return (
    <Box position="absolute" display="inline-flex" top={-30} right={2}>
      <StyledRatingContainer>
        <CircularProgress
          variant="determinate"
          size={50}
          thickness={4}
          value={rating}
        />
        <Box
          top={0}
          left={0}
          bottom={0}
          right={0}
          position="absolute"
          display="flex"
          alignItems="center"
          justifyContent="center"
        >
          <Typography
            variant="subtitle2"
            component="div"
            color="white"
            style={{ fontWeight: 600 }}
          >{`${Math.round(rating)}%`}</Typography>
        </Box>
      </StyledRatingContainer>
    </Box>
  );
};

export default Rating;
