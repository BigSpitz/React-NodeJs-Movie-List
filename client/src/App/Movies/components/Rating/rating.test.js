import React from 'react';
import Rating from './index';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

test('Rating renders', () => {
  const rating = render(<Rating />);
  expect(rating).toMatchSnapshot();
});

test('Rating with props shows percentage', () => {
  render(<Rating rating={100} />);
  expect(screen.getByText('100%')).toBeInTheDocument();
});

test('Rating with props float rounds number', () => {
  render(<Rating rating={79.8} />);
  expect(screen.getByText('80%')).toBeInTheDocument();
});
