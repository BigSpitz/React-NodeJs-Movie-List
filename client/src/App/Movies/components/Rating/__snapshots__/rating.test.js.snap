// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root MuiBox-root-1"
      >
        <div
          class="sc-bdnxRM wjeph"
        >
          <div
            aria-valuenow="0"
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
            role="progressbar"
            style="width: 50px; height: 50px; transform: rotate(-90deg);"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20"
                stroke-width="4"
                style="stroke-dasharray: 125.664; stroke-dashoffset: 125.664px;"
              />
            </svg>
          </div>
          <div
            class="MuiBox-root MuiBox-root-2"
          >
            <div
              class="MuiTypography-root MuiTypography-subtitle2"
              style="font-weight: 600;"
            >
              NaN%
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root MuiBox-root-1"
    >
      <div
        class="sc-bdnxRM wjeph"
      >
        <div
          aria-valuenow="0"
          class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
          role="progressbar"
          style="width: 50px; height: 50px; transform: rotate(-90deg);"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20"
              stroke-width="4"
              style="stroke-dasharray: 125.664; stroke-dashoffset: 125.664px;"
            />
          </svg>
        </div>
        <div
          class="MuiBox-root MuiBox-root-2"
        >
          <div
            class="MuiTypography-root MuiTypography-subtitle2"
            style="font-weight: 600;"
          >
            NaN%
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
